[Liu Fangrui] iP

LiuFangrui opened the pull request:
### DukeEx

> “Goals are dreams with deadlines.”  ― Diana Scharf

- text-based
- cater to up to 3 types of tasks
- autosaves based on every edit

Use DukeEx by following these simple steps:

1. Download [DukeEx](https://github.com/LiuFangrui/ip)
2. Run DukeEx
3. Start updating your daily tasks!
4. _Never miss a deadline again_ 👍🏻

Features:
- [x] Manages different types of tasks (e.g. Deadlines, Events, ...)
- [x] Search for Tasks by name
- [ ] Sort by the due date

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) {
        new Duke(DATA_FILE_PATH).run();
}

LiuFangrui committed:
Level-1

LiuFangrui committed:
Level-2

LiuFangrui committed:
Level-4

LiuFangrui committed:
A-TextUiTesting

LiuFangrui committed:
Level-5

LiuFangrui committed:
Level-6

LiuFangrui committed:
Updated A-TextUiTesting

LiuFangrui committed:
Merge branch 'master' of https://github.com/LiuFangrui/ip

# Conflicts:
#	src/main/java/Duke.java
#	text-ui-test/input.txt

LiuFangrui committed:
fixed corrupted files

LiuFangrui committed:
Level-7

LiuFangrui committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Duke.java

LiuFangrui committed:
working copy after merging Level 7

LiuFangrui committed:
Level-8

LiuFangrui committed:
A-MoreOOP

LiuFangrui committed:
removed unneeded dependencies

LiuFangrui committed:
created duke package

LiuFangrui committed:
A-Gradle

LiuFangrui committed:
rerouted package

LiuFangrui committed:
reroute package(1)

LiuFangrui committed:
A-JUnit

LiuFangrui committed:
A-JavaDoc

LiuFangrui committed:
A-CodingStandard

LiuFangrui committed:
Level-9

eesung00 commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+    private final String[] commandWords = new String[]{"list", "mark",
+            "unmark", "todo", "event", "deadline", "delete", "bye", "find"};
+
+    public void parse(String text) throws EmptyMessageException, InvalidCommandException {
```
Should all public methods have Javadoc comments except getters/setters? I noticed the same issue in several places too.

eesung00 commented on src/test/java/seedu/duke/TaskListTest.java:
```diff
@@ -0,0 +1,39 @@
+
+public class TaskListTest {
+    @Test
+    public void Test1(){
```
Will the naming of test methods be better if you follow the test methods naming standards?
```suggestion
    public void addTest_addTask_expectedBehaviour( )
```

eesung00 commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+package seedu.duke;
+
+import java.util.Scanner;
+import java.io.File;
+import java.io.IOException;
```
Perhaps these import statement can be in lexical order?

eesung00 commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+            super(msg);
+        }
+    }
+    public static class EmptyMessageException extends DukeException {
+        public EmptyMessageException() {
+            super("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+    }
+    public static class InvalidCommandException extends DukeException {
+        public InvalidCommandException() {
+            super("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
```
I like how you extends DukeException into more specific Exceptions! 👍

eesung00 commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+package seedu.duke;
```
All your classes and variables naming are intuitive and follows the coding standard. Good Job!

eesung00 reviewed (commented):
LGTM. Just some few nits to improve

ConradLew commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+    private static String DATA_FILE_PATH = "./Duke.txt";
+
+    private Storage storage;
+    private TaskList taskL;
```
Perhaps less "short-form" naming or plural naming such as `taskList` or `tasks` would be more appropriate here?

ConradLew commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+        this.parser = new Parser(this.taskL, this.ui, this.storage);
+    }
+
+    public void run() {
```
I noticed that there are missing method headers for many public methods in the codebase. Should method headers be added to all public methods?

ConradLew commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+    private Storage storage;
+
+    /**
+     * Instantiates a new Parser object
```
Perhaps `@param` tags could be added into the method headers when applicable?

ConradLew commented on src/main/java/seedu/duke/Ui.java:
```diff
@@ -0,0 +1,68 @@
+     */
+    public void displayList(ArrayList<Task> l) {
+        String result = "";
+        if (l.isEmpty()) {
```
I like how all your boolean-returning methods/boolean variables appropriately sound like booleans. It makes the code a lot easier to understand.

ConradLew commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+        }
+    }
+
+    private boolean checkCommand(String s, int i) {
```
Perhaps some comments or more intuitive name of what part of the command is being checked would be appropriate here?

ConradLew commented on src/main/java/seedu/duke/Storage.java:
```diff
@@ -0,0 +1,58 @@
+     * @param tasks the TaskList containing Task to be written to file
+     */
+    public void writeToFile(TaskList tasks) throws IOException {
+        //structure: command|1 (1 for mark, 0 for unmark)
```
I like how the comment is added for clarity and the SLAP principles followed in the following lines. Good job!

ConradLew reviewed (commented):
Overall, your code meets the coding standard with some nits on naming and documentation. Well done!

LiuFangrui committed:
JavaDoc Part 1

LiuFangrui committed:
JavaDoc part 2

LiuFangrui committed:
javadoc part 3

LiuFangrui committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

LiuFangrui committed:
JavaFx

LiuFangrui committed:
Duke Gui

LiuFangrui committed:
Added assertions to parser

Run time errors can occur when index is out of list range or
command to add task is in a wrong format

checked for delete index and add task input

LiuFangrui committed:
Task classes: extracted Task classes and Todo, Deadline, Event
subclasses

Todo, Deadline, Event have different behaviours

The different behaviors across Todo, Deadline, Event classes cause
deep nesting.

extracting Todo, Deadline, Event classes makes code more readable
and reduce deep nesting

Task class is reduced to a abstract parent class containing shared
functions between Todo, Deadline and Event classes

LiuFangrui committed:
Minor changes to javadoc

LiuFangrui committed:
Merge pull request #1 from LiuFangrui/branch-A-Assertions

branch-A-Assertions

LiuFangrui committed:
Merge branch 'branch-A-CodeQuality'

LiuFangrui committed:
C-Update

LiuFangrui committed:
A-UserGuide

LiuFangrui committed:
Incorrect Ui.png, corrected Ui.png.png to Ui.png

LiuFangrui committed:
updated storage to cater to all OS
